src/BenchmarksApps/MapAction/Program.cs

```diff
@@ -23,6 +25,29 @@
 
                 object Json() => new { message = "Hello, World!" };
                 endpoints.MapGet("/json", (Func<object>)Json);
+
+                // Parameterized plain-text endpoint
+                string SayHello(string name) => $"Hello, {name}!";
+                // With no filters
+                endpoints.MapGet("/hello/{name}", SayHello);
+                // With a filter that no-ops
+                endpoints.MapGet("/helloEmptyFilter/{name}", SayHello)
+                    .AddFilter((context, next) => next(context));
+                endpoints
+                    .MapGet("/helloFiltered/{name}", SayHello)
+                    .AddFilter((routeHandlerContext, next) => {
+                        var hasSingleParam = routeHandlerContext.MethodInfo.GetParameters().Length == 1;
+                        var isStringParam = routeHandlerContext.MethodInfo.GetParameters()[0].ParameterType == typeof(string);
+
+                        return (context) =>
+                        {
+                            if (hasSingleParam && isStringParam)
+                            {
+                                context.Parameters[0] = context.Parameters[0].ToUpperInvariant();
+                            }
+                            return next(context);
+                        };
+                    });
```

Comment:
What is this one for?

Reply:
I see parameter mutation?

Reply:
Yeah, we can. I was trying to do the base case (no filters), pass through case (filter that does nothing but invoke next), and a case with some logic. I guess what logic we want to test will depend on what we care about.